contracts/adapters/OptimisticAuctionRebalanceExtension.sol

```diff
@@ -0,0 +1,390 @@
+     */
+    function addAllowedAssets(address[] memory _assets) external onlyOperator {
+        _addAllowedAssets(_assets);
+    }
+
+    /**
+     * ONLY OPERATOR: Remove asset(s) so that it/they can't be traded to, wrapped to, or claimed
+     *
+     * @param _assets           Asset(s) to remove
+     */
+    function removeAllowedAssets(address[] memory _assets) external onlyOperator {
+        _removeAllowedAssets(_assets);
+    }
+
+    /**
+     * ONLY OPERATOR: Toggle useAssetAllowlist on and off. When false asset allowlist is ignored
+     * when true it is enforced.
+     *
+     * @param _useAssetAllowlist           Bool indicating whether to use asset allow list
+     */
+    function updateUseAssetAllowlist(bool _useAssetAllowlist) external onlyOperator {
+        _updateUseAssetAllowlist(_useAssetAllowlist);
+    }
+
+
+
+    /**
+    * @dev OPERATOR ONLY: sets product settings for a given set token
+    * @param _optimisticParams OptimisticRebalanceParams struct containing optimistic rebalance parameters.
+    * @param _rulesHash bytes32 containing the ipfs hash rules for the product.
+    */
+    function setProductSettings(
+        OptimisticRebalanceParams memory _optimisticParams,
+        bytes32 _rulesHash
+    )
+        external
+        onlyOperator
+    {
+        productSettings[setToken] = ProductSettings({
+            optimisticParams: _optimisticParams,
+            rulesHash: _rulesHash
+        });
+        emit ProductSettingsUpdated(setToken, setToken.manager(), _optimisticParams, _rulesHash);
+    }
+
+     /**
+     * @param _quoteAsset                   ERC20 token used as the quote asset in auctions.
+     * @param _oldComponents                Addresses of existing components in the SetToken.
+     * @param _newComponents                Addresses of new components to be added.
+     * @param _newComponentsAuctionParams   AuctionExecutionParams for new components, indexed corresponding to _newComponents.
+     * @param _oldComponentsAuctionParams   AuctionExecutionParams for existing components, indexed corresponding to
+     *                                      the current component positions. Set to 0 for components being removed.
+     * @param _shouldLockSetToken           Indicates if the rebalance should lock the SetToken.
+     * @param _rebalanceDuration            Duration of the rebalance in seconds.
+     * @param _positionMultiplier           Position multiplier at the time target units were calculated.
+    */
+    function proposeRebalance(
+        IERC20 _quoteAsset,
+        address[] memory _oldComponents,
+        address[] memory _newComponents,
+        AuctionExecutionParams[] memory _newComponentsAuctionParams,
+        AuctionExecutionParams[] memory _oldComponentsAuctionParams,
+        bool _shouldLockSetToken,
+        uint256 _rebalanceDuration,
+        uint256 _positionMultiplier
+    )
+        external
+        onlyAllowedAssets(_newComponents)
```

Comment:
should we move this modifier over to `startRebalance`? considering a staleness case where the asset allowlist is modified between `proposeRebalance` and `startRebalance` calls.

Reply:
Yes makes sense. Moved it over accordingly.